be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -425,107 +431,323 @@ Status HorizontalBetaRowsetWriter::_final_merge() {
     MonotonicStopWatch timer;
     timer.start();
 
-    auto schema = vectorized::ChunkHelper::convert_schema_to_format_v2(*_context.tablet_schema);
+    if (CompactionUtils::choose_compaction_algorithm(_context.tablet_schema->num_columns(),
+                                                     config::vertical_compaction_max_columns_per_group,
+                                                     _num_segment) == VERTICAL_COMPACTION) {
```

Comment:
`seg_iterators` does not contain the segment that has no rows.
`_num_segment` (`source_num`) should to be same with `seg_iterators`.

Reply:
changed.